src/network.js

```diff
@@ -102,27 +132,38 @@ class Network {
     }
   }
 
+  /**
+   * @private
+   * @param {PeerId} peerId
+   */
   _onPeerConnect (peerId) {
-    this.bitswap._onPeerConnected(peerId)
+    this._bitswap._onPeerConnected(peerId)
   }
 
+  /**
+   * @private
+   * @param {PeerId} peerId
+   * @returns {void}
+   */
   _onPeerDisconnect (peerId) {
-    this.bitswap._onPeerDisconnected(peerId)
+    this._bitswap._onPeerDisconnected(peerId)
   }
 
   /**
    * Find providers given a `cid`.
    *
    * @param {CID} cid
    * @param {number} maxProviders
-   * @param {Object} options
-   * @param {AbortSignal} options.abortSignal
-   * @returns {AsyncIterable<PeerInfo>}
+   * @param {Object} [options]
+   * @param {AbortSignal} [options.signal]
+   * @returns {AsyncIterable<Provider>}
    */
   findProviders (cid, maxProviders, options = {}) {
-    return this.libp2p.contentRouting.findProviders(
+    return this._libp2p.contentRouting.findProviders(
       cid,
       {
+        // TODO: Should this be a timeout options insetad ?
```

Comment:
@vasco-santos do you know if we should be passing `timeout` instead of `maxTimeout`